### ISSUE TITLE
Support for tracing high-level methods of Java Classes via patterns

### ISSUE THREAD

**Issue by dpnishant** [frida/frida-python#70]
We need to add support for tracing high-level methods of Java Classes via patterns i.e. the Android/Java equivalent of the existing switch, below:

`frida-trace -U "com.some.app" -m *[NSFile* write*:*]`

**Comment by dpnishant** [comment]
may be we should also rename `-m` to `-ObjC` so that when we have the Java support, a switch like `-Java` would be consistent and self-explanatory. Just my 2 cents.

**Comment by Piasy** [comment]
Any update on this issue?

**Comment by oleavr** [comment]
@Piasy Nobody is working on this as far as I can know. Are you volunteering? ;)

**Comment by dpnishant** [comment]
@Piasy In the meantime, why don't you give https://github.com/dpnishant/appmon a try? :)

**Comment by Piasy** [comment]
@dpnishant Great work! But why not support wildcard in method name of Android Tracer? btw, I've write a tiny Java program to generate js code to hook all methods in provided classes, I'll release it later :)

**Comment by oleavr** [comment]
@Piasy Would be awesome if you could implement that in [frida-trace](https://github.com/frida/frida-python/blob/master/src/frida/tracer.py). It's still a pretty simple program, less than 900 LOC, so shouldn't take long to understand it and be able to add this new feature to it.

**Comment by Piasy** [comment]
Sure, I'll try it.

**Comment by Piasy** [comment]
@oleavr Sorry sir, I'm not very familiar with python, so I can't finish this job in a short time, and I don't have enough time to dive into this job. But I've released the Java program I wrote to generate hook script, hope it could help others, https://github.com/Piasy/FridaAndroidTracer.

**Comment by dpnishant** [comment]
@Piasy Check out this update https://github.com/dpnishant/appmon/commit/bf3cee9e3676ecb1f446c25aa7f6573a85794ace 😊

**Comment by Piasy** [comment]
@dpnishant Awesome!

**Comment by XhstormR** [comment]
Yes, it would be great if frida-trace could track the java method directly.

**Comment by bitterbit** [comment]
As of version 12.10 and frida-tools 8.0 frida-trace supports Android 🎉 
https://frida.re/news/2020/06/29/frida-12-10-released/

**Comment by Cyph3rex** [comment]
wow , this thread has been here for almost 6 years , and I'm looking for the same thing , a way to trace methods

**Comment by XhstormR** [comment]
Does frida currently support tracking methods on the JVM?
